server/routes/referrals/referralFormPresenter.ts

```diff
@@ -148,48 +153,127 @@ export default class ReferralFormPresenter {
         ],
       }
     }
-    const checkYourAnswers: ReferralFormSingleListSectionPresenter = {
-      type: 'single',
-      title: 'Check your answers',
-      number: '3',
-      status: this.calculateStatus(this.sectionValues.checkYourAnswers, referralDetails.status),
-      tasks: [
-        {
-          title: 'Check your answers',
-          url: this.calculateTaskUrl('check-answers', this.taskValues.furtherInformation),
-        },
-      ],
-    }
-    return [reviewServiceUserInformation, referralDetails, checkYourAnswers]
-  }
 
-  private calculateTaskUrl(url: string | null, displayCriteria: DraftReferralValues): string | null {
-    const everyFieldHasValues = displayCriteria.every(field => {
-      if (field === null) {
-        return false
+    private buildCohortReferralDetailsSection(
+      selectServiceCategoriesSection: ReferralFormSingleListSectionPresenter
+    ): ReferralFormMultiListSectionPresenter {
+      return {
+        type: 'multi',
+        title: 'Add intervention referral details',
+        number: '3',
+        status: this.calculateStatus(
+          this.sectionValues.serviceCategoryReferralDetails,
+          selectServiceCategoriesSection.status!
+        ),
+        taskListSections: [
+          {
+            tasks: [
+              {
+                title: `Confirm the relevant sentence for the intervention referral`,
+                url: this.calculateTaskUrl('relevant-sentence', this.taskValues.cohortServiceCategories),
+              },
+            ],
+          },
+        ]
+          .concat(
+            this.intervention.serviceCategories
+              .filter(serviceCat => {
+                if (this.referral.serviceCategoryIds != null) {
```

Comment:
nit: this should be `!==` to avoid unexpected null coalescing.

Reply:
unless that's what we want here.

Reply:
thanks for the spot